src/screens/DiscoveryScreenV2/fragment/UsersFragment.js

```diff
@@ -375,7 +375,7 @@ const UsersFragment = ({
         data={peopleYouMightKnowData || []}
         renderItem={renderItem}
         keyExtractor={(item, index) => index.toString()}
-        onEndReached={() => fetchData()}
+        onEndReached={() => (initialAccordionData.length > 0 ? fetchData() : null)}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The change made here is a good one as it adds a condition to check if `initialAccordionData` has any elements before calling `fetchData()`. This can prevent unnecessary calls to `fetchData()` when there's no data to fetch, improving performance. However, I would suggest using the `Array.isArray()` method to ensure that `initialAccordionData` is indeed an array and not `null` or `undefined`, which could cause an error when trying to access the `length` property.

```diff
- onEndReached={() => (initialAccordionData.length > 0 ? fetchData() : null)}
+ onEndReached={() => (Array.isArray(initialAccordionData) && initialAccordionData.length > 0 ? fetchData() : null)}
```
This way, we're ensuring that `initialAccordionData` is an array before checking its length, providing an extra layer of safety against potential errors.


<!-- This is an auto-generated comment by OSS CodeRabbit -->